### PR TITLE
issue325

### PR DESCRIPTION
**Pull request by Listener430**
This is related to both #325 and #326
In order to test using Docker as in naked start:
1. Add this [prest.toml](https://gist.github.com/Listener430/9eb6a21cdafd7155e1c8d35e53609e92) to root
2. Had to add line 19 into Dockerfile:
`COPY --from=builder /workspace/prest.toml /app/`
3. Create table or add migrations
`docker-compose exec postgres psql -d prest -U postgres -c "create table person(name varchar(250), age integer, adr varchar(250))"`

### CODE REVIEW

File: adapters/postgres/postgres.go [hunk 1]
```diff
@@ -355,35 +355,48 @@ func (adapter *Postgres) SetByRequest(r *http.Request, initialPlaceholderID int)
 		err = ErrBodyEmpty
 		return
 	}
+	allowedFields := fieldsByPermission(table, op)
+	if len(allowedFields) == 0 {
+		err = ErrBodyEmpty
```
Comment (avelino) [inline adapters/postgres/postgres.go:360]: Added lines #L360 - L362 were not covered by tests

File: adapters/postgres/postgres.go [hunk 2]
```diff
@@ -355,35 +355,48 @@ func (adapter *Postgres) SetByRequest(r *http.Request, initialPlaceholderID int)
...
-		switch reflect.ValueOf(value).Kind() {
-		case reflect.Interface:
-			values = append(values, formatters.FormatArray(value))
-		case reflect.Map:
-			jsonData, err := json.Marshal(value)
-			if err != nil {
-				log.Errorln(err)
+		if k || set[key] {
+			if chkInvalidIdentifier(key) {
+				err = errors.Wrap(ErrInvalidIdentifier, "Set")
```
Comment (avelino) [inline adapters/postgres/postgres.go:373]: Added lines #L373 - L374 were not covered by tests

**Review by avelino — CHANGES_REQUESTED** [review]
we had a loss of test coverage, it is important to raise the coverage level (mainly by changing the function signature)

**Comment by Listener430** [comment]
seems I wanted to sync my branch and created a commit ..

File: adapters/adapter.go [hunk 3]
```diff
@@ -71,7 +71,7 @@ type Adapter interface {
 	OrderByRequest(r *http.Request) (values string, err error)
 	PaginateIfPossible(r *http.Request) (paginatedQuery string, err error)
 	ParseBatchInsertRequest(r *http.Request) (colsName string, colsValue string, values []interface{}, err error)
-	ParseInsertRequest(r *http.Request) (colsName string, colsValue string, values []interface{}, err error)
+	ParseInsertRequest(r *http.Request, table string, op string) (colsName string, colsValue string, values []interface{}, err error)
```
Comment (arxdsilva) [inline adapters/adapter.go:74]: we cant break that signature now, this would impact too many people

File: adapters/adapter.go [hunk 4]
```diff
@@ -87,7 +87,7 @@ type Adapter interface {
 	SchemaTablesWhere(requestWhere string) (whereSyntax string)
 	SelectFields(fields []string) (sql string, err error)
 	SelectSQL(selectStr string, database string, schema string, table string) string
-	SetByRequest(r *http.Request, initialPlaceholderID int) (setSyntax string, values []interface{}, err error)
+	SetByRequest(r *http.Request, initialPlaceholderID int, table string, op string) (setSyntax string, values []interface{}, err error)
```
Comment (arxdsilva) [inline adapters/adapter.go:90]: same here

File: adapters/adapter.go (hunk 3 above)
Comment (Listener430) [reply adapters/adapter.go:74]: Ok, please close this PR than, I will create new one with another proposed solution

**Comment by Listener430** [comment]
Closing PR, would think about another implementation

**Pull request closed without merging**